crates/ra_ide/src/hover.rs

```diff
@@ -2410,4 +2410,103 @@ fn func(foo: i32) { if true { <|>foo; }; }
             ]
             "###);
     }
+
+    #[test]
+    fn infer_closure_arg() {
```

Comment:
oh, I actually missed that these are hover tests. Tests for type inference should mostly be in `ra_hir_ty` :slightly_smiling_face: I'd probably put them in `tests/traits.rs`. While we're at it, we can turn them into snapshot tests to cover both the inferred return value and arguments.